system/services/admin/DataManagerService.cfc

```diff
@@ -227,6 +238,47 @@ component output="false" singleton=true {
 
 		return result;
 	}
+	public struct function batchSaveFieldChanges(
+		  required string  objectName
+		, required string  sourceIds
+		, required string  updateValue
+		, required string  updateField
+		, required string  DataColumn
+		, required string  attributeRelationship
+		,          string  overwrite      = "append"
+	) output=false {
+
+		for( sourceID in sourceIds ) {
+			if( attributeRelationship == "many-to-many" ) {
+				multiSelectedValue = updateValue;
+				if( overwrite != "overwrite") {
+					var previousData = _getPresideObjectService().getDeNormalizedManyToManyData( objectName = objectName, id = sourceID );
+					data[ sourceID ] = ListRemoveDuplicates( listAppend( multiSelectedValue, previousData[ DataColumn ] ) );
+				}else{
+					data[ sourceID ] =  multiSelectedValue;
+				}
+					result.multiSelect   = _getPresideObjectService().syncManyToManyData(  sourceObject   = objectName
+											                                 , sourceProperty = DataColumn
+											                                 , sourceId       = sourceID
+											                                 , targetIdList   = data[ sourceID ] );
+			}else {
+				if ( attributeRelationship  == "many-to-one"){
+					singleSelectedValue     = updateValue;			
+					formData[ DataColumn ]  = singleSelectedValue;
+				} else {
+					singleSelectedValue     = updateValue;
+					formData[ updateField ] = singleSelectedValue;
+				}
+				
+				result.singleSelect = _getPresideObjectService().updateData( objectName  = objectName 
+																            , data       = formData 
+																            , id         = sourceId  );
+				structClear(formData);
```

Comment:
Rather than this, it would be cleaner to do:

```
for( var sourceId in sourceIds ) {
    var formData = {};
}
```

This also fixes the fact that `formData` is not `var` scoped.